feat: replace netrc with json

alvarosabu opened the pull request:
This PR replaces the previous approach to save login credentials on `~/.netrc` file  to a more comfortable `~/.storyblok/credentials.json` simplifying the code required and avoiding compatibility issues with other python libraries

> [!WARNING]  
> This PR doesn't include multi-region sessions. This will be done  in a separate PR.

Closes #137 

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Other (please describe):

## How to test this PR

Login credentials now should be saved at `~/.storyblok/credentials.json` with no disruption on the login/logout functionality

alvarosabu committed:
feat: replace netrc with json

alvarosabu committed:
fix: clean linter

alvarosabu committed:
feat: updated tests to remove netrc

edodusi commented on src/commands/components/actions.ts:
adding an empty file here?

edodusi commented on src/creds.ts:
```diff
@@ -1,260 +1,65 @@
-  }
-}
-
-export async function isAuthorized() {
```
this and `removeAllNetrcEntries` are imported and used in `src/commands/logout/index.ts`

edodusi commented on src/index.ts:
```diff
@@ -31,8 +31,11 @@ program.command('test').action(async () => {
+
+    await initializeSession()
+
+    console.log(state)
```
do we need this log here?

alvarosabu replied:
This command is for quick testing purposes, is gonna be removed on the final release, consider it as a playground for now

edodusi replied:
ok!

edodusi commented on src/session.ts:
```diff
@@ -29,9 +29,9 @@ function createSession() {
-    const machines = await getNetrcCredentials()
-    const creds = getCredentialsForMachine(machines, machineName)
+    // If no environment variables, fall back to .storyblok/credentials.json
+    const machines = await getCredentials()
```
in a clean environment, launching a `login` will fail because `getCredentials()` if it doesn't find the `credentials.json` throws a filesystem error (`handleFileSystemError`) that should be catch and handled here

edodusi replied:
fixed

alvarosabu committed:
chore: delete unused path

alvarosabu committed:
feat: refactor credential management to handle non-existent files and update session handling

- Updated `getCredentials` to create a credentials file if it does not exist, returning an empty object.
- Changed `removeCredentials` to accept a region code instead of a machine name, using the region to determine the machine name.
- Refactored `persistCredentials` to use region code directly.
- Updated tests to reflect changes in credential management and session handling.

alvarosabu committed:
Merge branch 'next' into bugfix/replace-netrc-with-json

alvarosabu committed:
chore: remove leftovers of netrc
